### PR TITLE
feat/order age from creation timestamp

### PR DESCRIPTION
**Pull request by aarmoa**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
Change in the implementation of the order age calculation to stop using the timestamp that is part of the order_id and start using the new creation_timestamp field in the orders.

**Tests performed by the developer**:
Added unit tests.
Manually tested using a PMM strategy connected to paper trade Binance spot connector.

**Tips for QA testing**:
- Check status command for the strategies report age correctly.
- Check age in trade events stored in the DB and in CSV files is correct.

[ch21140]

### CODE REVIEW

File: hummingbot/strategy/utils.py [hunk 1]
```diff
@@ -1,15 +1,18 @@
 import time
+from typing import Optional
 
 from hummingbot.core.data_type.limit_order import LimitOrder
 
 
-def order_age(order: LimitOrder) -> float:
+def order_age(order: LimitOrder, current_time: Optional[float] = None) -> int:
     """
     Get the age of a limit order in second, not applicable to paper trade orders
```
Comment (petioptrv) [inline hummingbot/strategy/utils.py:9]:
```suggestion
    Get the age of a limit order in second.
```

File: hummingbot/strategy/hedge/hedge.pyx [hunk 2]
```diff
@@ -120,13 +108,13 @@ cdef class HedgeStrategy(StrategyBase):
             object order_size_quantum = market.c_get_order_size_quantum(market_pair.trading_pair, quantized_order_amount)
 
         for o in active_orders:
-    
-            if isnan(order_age(o)):
+            order_age = order_age(0, self._current_timestamp)
```
Comment (petioptrv) [inline hummingbot/strategy/hedge/hedge.pyx:111]:
```suggestion
            order_age = order_age(o, self._current_timestamp)
```

**Review by petioptrv — CHANGES_REQUESTED** [review]

**Comment by PtrckM** [comment]
Update: Confirmed debugging messages are now remove, changes are look good and behaving as expected. We will wait for the requested changes to be pushed before approving.

**Review by PtrckM — APPROVED** [review]
Tested and approved by QA team.
Test Summary:
    - Create and start a PMM strategy using AscendEX, Binance, and Binance_paper_trade connector
    - Run status --live and monitor the age of each order created: Confirmed that each order gets cancelled according to the specified order refresh time
    - Filled event: Ok
    - Run history --verbose to check the order_age
    - Compare the timestamp saved in the CSV and sqlite file (filled table): Match
    - Compare the order age of the filled order using the data on history vs order age in the csv file and using the creation timestamp and filled timestamp in the sqlite file: Order age matched
    - Create an Avellaneda MM strategy, Liquidity Mining strategy, Perpetual MM, Aroon Osicillator and Hedge strategy then repeat the steps: Confirm that order age recorded in the history, CSV, and Sqlite file are matched
    - Confirmed that all exchanges that support paper trade working as expected with order age.

**Review by petioptrv — APPROVED** [review]
Changes LGTM!

**Comment by rxlxrxsx** [comment]
Recent changes still LGTM. I didn't submit an approval as it already has enough. Tests performed were same as above.

**Comment by aarmoa** [comment]
Closing this PR after creating an equivalent one in the Hummingbot repository: https://github.com/hummingbot/hummingbot/pull/5143

**Pull request closed without merging**